tests/test_utils.py

```diff
@@ -164,6 +164,241 @@ def test_list_timeframes(mocker, capsys):
     assert re.search(r"^1d$", captured.out, re.MULTILINE)
 
 
+def test_list_markets(mocker, markets, capsys):
+
+    api_mock = MagicMock()
+    api_mock.markets = markets
+    patch_exchange(mocker, api_mock=api_mock)
+
+    # Test with no --config
+    args = [
+        "list-markets",
+    ]
+    with pytest.raises(OperationalException,
+                       match=r"This command requires a configured exchange.*"):
+        start_list_markets(get_args(args), False)
+
+    # Test with --config config.json.example
+    args = [
+        '--config', 'config.json.example',
+        "list-markets",
+        "--print-list",
+    ]
+    start_list_markets(get_args(args), False)
+    captured = capsys.readouterr()
+    assert ("Exchange Bittrex has 8 active markets: "
+            "BLK/BTC, BTT/BTC, ETH/BTC, ETH/USDT, LTC/USD, LTC/USDT, TKN/BTC, XLTCUSDT.\n"
+            in captured.out)
+
+#    # Test with --exchange
+#    args = [
+#        "list-markets",
+#        "--exchange", "binance"
+#    ]
+#    start_list_markets(get_args(args), False)
+#    captured = capsys.readouterr()
+#    assert re.match("Exchange Binance has 8 active markets:\n",
+#                    captured.out)
```

Comment:
Is this intentionally commented?

Reply:
Uncommented it (slightly changed), it fails and I do not see the reason for it (does not fetch the exchange name). The option works fine in the subcommand when run standalone...

Reply:
The exchange name is taken from the _api object, which is mocked ... so you need to change the mock (and change it back afterwards).


I've fixed this test now - have a look at my changes ... 

For simplicity, i used `patch_exchange()` again (and again) - but technically we could also extract  `mocker.patch('freqtrade.exchange.Exchange.name', PropertyMock(return_value=id.title()))` from that method and mock it directly.

Reply:
Thanks, but it;'s strange why does the similar test for list_timeframes works without this...

Reply:
Because `list-timeframes` uses the value from the configuration (`{config['exchange']['name']}` as output, and this command uses `exchange.name` ...

Reply:
should this also be fixed in the same manner (or list-timeframes changed instead)?

Reply:
well ... if we change list-timeframes to use exchange.name, then yes, we must change it in this way.
an alternative would to remove my fix for `list-pairs`, and use `config['exchange']['name']` instead ... 

Personally, i prefer using exchange.name, since it Capitalizes the name correctly - but i'm not sure if i would want to touch `list-timeframes` as part of this pr ...

Reply:
okay, got it (the difference, still asleep in the meantime), thanks 👍